src/Catalyst.Core.Modules.Dfs/DfsService.cs

```diff
@@ -0,0 +1,663 @@
+
+            var peerManager = new PeerManager
+            {
+                SwarmService = swarm
+            };
+            await peerManager.StartAsync().ConfigureAwait(false);
+            _stopTasks.Add(async () => { await peerManager.StopAsync().ConfigureAwait(false); });
+
+            // Start the primary services.
+            var tasks = new List<Func<Task>>
+            {
+                async () =>
+                {
+                    var bitSwap = BitSwapService;
+                    _stopTasks.Add(async () => await bitSwap.StopAsync().ConfigureAwait(false));
+                    await bitSwap.StartAsync().ConfigureAwait(false);
+                },
+                async () =>
+                {
+                    var dht = DhtService;
+                    _stopTasks.Add(async () => await dht.StopAsync().ConfigureAwait(false));
+                    await dht.StartAsync().ConfigureAwait(false);
+                },
+                async () =>
+                {
+                    var ping = PingService;
+                    _stopTasks.Add(async () => await ping.StopAsync().ConfigureAwait(false));
+                    await ping.StartAsync().ConfigureAwait(false);
+                },
+                async () =>
+                {
+                    var pubsub = PubSubService;
+                    _stopTasks.Add(async () => await pubsub.StopAsync().ConfigureAwait(false));
+                    await pubsub.StartAsync().ConfigureAwait(false);
+                }
+            };
+
+            Log.Debug("waiting for services to start");
+            await Task.WhenAll(tasks.Select(t => t())).ConfigureAwait(false);
+
+            // Starting listening to the swarm.
+            var json = await ConfigApi.GetAsync("Addresses.Swarm").ConfigureAwait(false);
+            var numberListeners = 0;
+            foreach (string a in json)
+            {
+                try
+                {
+                    await swarm.StartListeningAsync(a).ConfigureAwait(false);
+                    ++numberListeners;
+                }
+                catch (Exception e)
+                {
+                    Log.Warning($"Listener failure for '{a}'", e);
+                }
+            }
+
+            if (numberListeners == 0)
+            {
+                Log.Error("No listeners were created.");
+            }
+            
+            // Now that the listener addresses are established, the discovery 
+            // services can begin.
+            MulticastService multicast = null;
+            if (!Options.Discovery.DisableMdns)
+            {
+                multicast = new MulticastService();
+#pragma warning disable CS1998
+                _stopTasks.Add(async () => multicast.Dispose());
+#pragma warning restore CS1998
+            }
+
+            var autodialer = new AutoDialer(swarm)
+            {
+                MinConnections = Options.Swarm.MinConnections
+            };
+#pragma warning disable CS1998
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not suppress issues.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)